core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -528,6 +544,18 @@ public boolean isLegacyDefault() {
             } else {
                 core = null;
             }
+
+            JSONArray w = o.optJSONArray("warnings");
+            if (w != null) {
+                for (int i = 0; i < w.size(); i++) {
+                    try {
+                        warnings.add(new Warning(w.getJSONObject(i)));
+                    } catch (JSONException ex) {
+                        LOGGER.log(Level.WARNING, "Failed to parse JSON for warning", ex);
```

Comment:
Would be useful to add some info about the location of the JSON node. E.g. "warning #XXX"